pkg/parse/parse.go

```diff
@@ -555,7 +564,7 @@ func PlatformsFromOptions(c *cobra.Command) (platforms []struct{ OS, Arch, Varia
 		if err != nil {
 			return nil, fmt.Errorf("unable to parse platform: %w", err)
 		}
-		if os != "" || arch != "" || variant != "" {
+		if os != runtime.GOOS || arch != runtime.GOARCH || variant != "" {
```

Comment:
Do the changes in this function affect test results?

Reply:
We are now defaulting the OS and ARCH and only care if the user changes the value to something different then the default

before on an linux/amd64 machine `--platform=linux/amd64 --os=linux --arch=amd64` blue up, now it is allowed since it ultimately does what the user wanted.

If the user does
`--platform=linux/amd64 --os=linux --arch=arm64` then the command will fail

Reply:
It doesn't check that the `--platform` and `--arch`/`--os` values agree.  And what's the benefit of allowing _some_ combinations of the flags?  Why encourage people with "well, sometimes, it's allowed, in newer versions"?

Reply:
It checks if the --arch or --os changed values from the default.
Similar to --arch="" and --os="" in the old version would have been ignored.

I think we are arguing about a nothing burger, and will relent to make you happy, even though this is the corner case of all corner cases.